### PR TITLE
Alerting Enhancements: Alerting Comments (Experimental)

### PR DESCRIPTION
**Pull request by toepkerd**
*Issue #, if available:*

*Description of changes:*

*CheckList:*
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/BucketLevelMonitorRunner.kt [hunk 1]
```diff
@@ -325,11 +335,34 @@ object BucketLevelMonitorRunner : MonitorRunner() {
                         continue
 
                     val actionCtx = triggerCtx.copy(
-                        dedupedAlerts = dedupedAlerts,
+                        dedupedAlerts = dedupedAlerts.map {
+                            val dedupedAlertsComments = CommentsUtils.getCommentsForAlertNotification(
```
Comment (AWSHurneyt) [inline alerting/src/main/kotlin/org/opensearch/alerting/BucketLevelMonitorRunner.kt:339]: As discussed offline, `getCommentsForAlertNotification` is being called for each alert in the `dedupedAlertsComments`, `newAlerts`, and `completedAlerts` lists. Since a notification from a bucket monitor can easily include dozens of alerts, that would mean dozens of search requests. @toepkerd is investigating consolidating the searchRequests for comments across all the lists into a single/batched request.

File: alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt [hunk 2]
```diff
@@ -684,6 +685,8 @@ class AlertService(
         val alertsIndex = dataSources.alertsIndex
         val alertsHistoryIndex = dataSources.alertsHistoryIndex
 
+        val commentsToDeleteIDs = mutableListOf<String>()
```
Comment (eirsep) [inline alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt:688]: commentIdsToDelete?

File: alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt [hunk 3]
```diff
@@ -684,6 +685,8 @@ class AlertService(
         val alertsIndex = dataSources.alertsIndex
         val alertsHistoryIndex = dataSources.alertsHistoryIndex
 
+        val commentIDsToDelete = mutableListOf<String>()
```
Comment (eirsep) [inline alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt:688]: nit:camelcase

File: alerting/src/main/resources/org/opensearch/alerting/comments/alerting_comments.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
+{
+  "dynamic": "false",
+  "properties": {
```
Comment (AWSHurneyt) [inline alerting/src/main/resources/org/opensearch/alerting/comments/alerting_comments.json:3]: Should these fields be in a nested `comment` object?

**Review by AWSHurneyt — APPROVED** [review]

**Comment by riysaxen-amzn** [comment]
let's add some more tests to verify all the behavior and edge cases.

**Comment by riysaxen-amzn** [comment]
rest lgtm , approving assuming CI will pass. But make sure CI passes before merging

**Review by riysaxen-amzn — APPROVED** [review]

**Comment by amsiglan** [comment]
> rest lgtm , approving assuming CI will pass. But make sure CI passes before merging

We will fix the CIs in follow up PR. @toepkerd is working on it

**Comment by toepkerd** [comment]
CIs have been fixed and merged in this followup PR: #1574
